lib/fluent/plugin/output.rb

```diff
@@ -326,6 +326,7 @@ def start
           @buffering = prefer_buffered_processing
           if !@buffering && @buffer
             @buffer.terminate # it's not started, so terminate will be enough
+            @buffer = nil
```

Comment:
It seems error prune.
Using configured value is better for detecting output plugin is un-buffered or buffered.